Shared/Extensions/URLExtensions.swift

```diff
@@ -148,9 +148,12 @@ extension URL {
 
     public func getQuery() -> [String: String] {
         var results = [String: String]()
-        let components = URLComponents(url: self, resolvingAgainstBaseURL: false)
 
-        guard let queryItems = components?.queryItems else { return results }
+        guard let components = URLComponents(url: self, resolvingAgainstBaseURL: false),
+              let queryItems = components.percentEncodedQueryItems
```

Comment:
Hmm you want a garbled url with query params? 🤔 

Basically `queryItems` will get a query with decoded params whereas the `percentEncodedQueryItems` will keep the encoded URL

Ex: `https://www.gocal.com/sessionrestore?url=https%3A%2F%2Fwww.similarweb.com%2Ftop-websites%2Fcomputers-electronics-and-technology%2Fsearch-engines%2F`

`components.queryItems`: 
[{name "url", value "https%3A%2F%2Fwww.similarweb.com%2Ftop-websites%2Fcomputers-electronics-and-technology%2Fsearch-engines%2F"}]

`components.percentEncodedQueryItems`
[{name "url", value "https://www.similarweb.com/top-websites/computers-electronics-and-technology/search-engines/"}]

Reply:
I can see why we might keep the percent-encoding but this gets used in NavigationRouter. Have we looked into the consequences it has if we change it?

Reply:
Based on Gutleys comments in the ticket I believe this is just going back to the behaviour we had before it was broken. When the method was updated to use url components it broke the handling of urls that have percent encoding. I'll take a look back at the old code just to make doubly sure.

Reply:
Yeah, code-wise this looks fine but being cautious of introducing a bug in NavigationRouter and in OpenSearchEngine which is where it is used. 

R+ for the changes in general

Reply:
Yeah, looking back at the old code it literally just separated the parameters based on & and the key value pairs within based on =, so this should fix the underlying issue and shouldn't regress anything else.